custom_components/powercalc/power_profile/loader/remote.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, hass: HomeAssistant) -> None:
 
     async def initialize(self) -> None:
         self.library_contents = await self.load_library_json()
-        self.last_update_time = self.get_last_update_time()
+        self.last_update_time = await self.get_last_update_time()
```

Comment:
You can do like this: `await self.hass.async_add_executor_job(self.get_last_update_time)`
And keep `get_last_update_time` sync method. Than we also don't block on `os.path.exists` which is also done in this function.